lib/AktiveMerchant/Billing/Gateways/Trexle.php

```diff
@@ -0,0 +1,442 @@
+<?php
+
+/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */
+
+namespace AktiveMerchant\Billing\Gateways;
+
+use AktiveMerchant\Billing\Interfaces as Interfaces;
+use AktiveMerchant\Billing\Gateway;
+use AktiveMerchant\Billing\CreditCard;
+use AktiveMerchant\Billing\Response;
+use AktiveMerchant\Common\Options;
+use AktiveMerchant\Http\RequestInterface;
+
+/**
+ * Integration of Trexle gateway.
+ *
+ * Basic usage:
+ *
+ *  use AktiveMerchant\Billing\CreditCard;
+ *
+ *  $options = ['secret_key' => <yoursecretkey>];
+ *  $gateway = AktiveMerchant\Billing\Base::gateway('trexle', $options);
+ *  // Setup customer token or card token or card data.
+ *  $card = new CreditCard(['token' => 'cus_xxxxxxx']); // Customer token
+ *  $card = new CreditCard(['token' => 'card_xxxxxxx']); // Card token
+ *  $card = new CreditCard([
+ *        "first_name" => "John",
+ *        "last_name" => "Doe",
+ *        "number" => "4242424242424242",
+ *        "month" => "01",
+ *        "year" => "17",
+ *        "verification_value" => "123"
+ *  ]); // Card data
+ *
+ *  // Complete purchase
+ *  $response = $gateway->purchase(1000, $card);
+ *
+ *  // Get unique reference from transaction for future use.
+ *  $authorization = $response->authorization();
+ *
+ * @author Andreas Kollaros <andreas@larium.net>
```

Comment:
It's actually your own code for Pin. Trexle API is identical, so I preferred to keep your name.
But if you insist, sure! 👍